lib/Compilation.js

```diff
@@ -1118,26 +1118,31 @@ BREAKING CHANGE: Asset processing hooks in Compilation has been merged into a si
 	_processModuleDependencies(module, callback) {
 		const dependencies = new Map();
 
+		/**
+		 * @type {Array<{factory: ModuleFactory, dependencies: Dependency[], originModule: Module|null}>}
+		 */
 		const sortedDependencies = [];
 
 		let currentBlock = module;
 
 		let factoryCacheKey;
 		let factoryCacheValue;
 		let factoryCacheValue2;
-		let listCacheKey;
+		let listCacheKey1;
+		let listCacheKey2;
 		let listCacheValue;
 
 		const processDependency = dep => {
 			this.moduleGraph.setParents(dep, currentBlock, module);
 			const resourceIdent = dep.getResourceIdentifier();
+			const category = dep.category;
```

Comment:
Instead of adding an additional layer of maps, better concat category and resource identifier. That's better for performance (and this is a pretty hot code path).

```suggestion
			const resourceIdent = `${dep.category} ${dep.getResourceIdentifier()}`;
```